acceptancetests/deploy_stack.py

```diff
@@ -570,6 +571,40 @@ def tear_down(self, has_controller):
             self.tear_down_client.kill_controller(check=True)
 
 
+class ExistingController:
+    """A Controller strategy where the controller is already present.
+
+    Intended for use with BootstrapManager.
+    """
+
+    def __init__(self, client, tear_down_client, juju_data_dir, env):
```

Comment:
Please include a docstring outlining what the args are expected to be here.
Seems a bit odd to set the client.env.environment to something else passed in, can you clarify please.
juju_data_dir is declared but un-used, not sure if you have plans for that?